client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -38,7 +41,10 @@ public void call() {
             eventListener.expectOnlyCreatedEvent(1, cache(cacheName));
          }
       });
-      detectThreadLeaks(".*Client-Listener-ClientListenerLeakTest-.*");
+      WaitFor
+              .condition(() -> listThreadsMatchingRegexp(".*Client-Listener-ClientListenerLeakTest-.*").size() == 0)
+              .atMost(10, TimeUnit.MILLISECONDS)
+              .await();
```

Comment:
So we wait 5 seconds for the threads to (hopefully) terminate in ClientListenerNotifier.stop. And then we wait 10 milliseconds more before triggering the alarm? I'd rather fix this by interrupting those threads in ClientListenerNotifier.stop.